test/assignment.coffee

```diff
@@ -449,7 +449,7 @@ test "#1591, #1101: splatted expressions in destructuring assignment must be ass
 
 test "#1643: splatted accesses in destructuring assignments should not be declared as variables", ->
   nonce = {}
-  accesses = ['o.a', 'o["a"]', '(o.a)', '(o.a).a', '@o.a', 'C::a', 'C::', 'f().a', 'o?.a', 'o?.a.b', 'f?().a']
```

Comment:
Can you explain why it’s safe to remove `C::`? This compiles to `C.prototype` in both this PR and CoffeeScript 1.x. I see that the test uses this with a `new class`, so presumably that’s why this test fails now; but why does it fail?